tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -2979,4 +2979,388 @@ public function testEnsurePrimaryKeyBeingReadFromTableWhenLoadingBelongsToManyRe
+        $method = $this->_getProtectedMethod($marshall, '_mergeTranslations');
+
+        $errors = [];
+        $data = ['en' => ['title' => 'Hi'], 'es' => ['title' => 'hola']];
+        $args = [
+            null,
+            $data,
+            &$errors
+        ];
+
+        $returnValue = [
+            [
+                'en' => null,
+                'es' => null
+            ],
+            []
+        ];
+        $mockTable->expects($this->once())
+            ->method('mergeTranslations')
+            ->with(
+                $this->isNull(),
+                $this->equalTo($data),
+                $this->isInstanceOf('Cake\ORM\Marshaller'),
+                $this->equalTo([])
+            )
+            ->will($this->returnValue($returnValue));
+
+        $result = $method->invokeArgs($marshall, $args);
+
+        $this->assertEquals($returnValue[0], $result);
+        $this->assertEmpty($errors);
+    }
+
+    /**
+     * Test Merge translation with errors, add to `$errors` parameter
+     *
+     * @return void
+     */
+    public function testMergeTranslationsWithErrors()
+    {
+        // Mocks
+        $mockBehavior = $this->_getMockBehaviors();
+        $mockTable = $this->_getMockTable(['mergeTranslations'], ['behaviors' => $mockBehavior]);
+
+        $marshall = new Marshaller($mockTable);
+        $method = $this->_getProtectedMethod($marshall, '_mergeTranslations');
+
+        $errors = [];
+        $data = ['en' => ['title' => 'Hi'], 'es' => ['title' => 'hola']];
+        $args = [
+            null,
+            $data,
+            &$errors
+        ];
+
+        $returnValue = [
+            [
+                'en' => null,
+                'es' => null
+            ],
+            [
+                'en' => []
+            ]
+        ];
+        $mockTable->expects($this->once())
+            ->method('mergeTranslations')
+            ->with(
+                $this->isNull(),
+                $this->equalTo($data),
+                $this->isInstanceOf('Cake\ORM\Marshaller'),
+                $this->equalTo([])
+            )
+            ->will($this->returnValue($returnValue));
+
+        $result = $method->invokeArgs($marshall, $args);
+
+        $this->assertEquals($returnValue[0], $result);
+        $this->assertArrayHasKey('_translations', $errors);
+    }
+
+    /**
+     * Test Merge translation when return null value
+     *
+     * @return void
+     */
+    public function testMergeTranslationsReturnNull()
+    {
+        // Mocks
+        $mockBehavior = $this->_getMockBehaviors();
+        $mockTable = $this->_getMockTable(['mergeTranslations'], ['behaviors' => $mockBehavior]);
+
+        $marshall = new Marshaller($mockTable);
+        $method = $this->_getProtectedMethod($marshall, '_mergeTranslations');
+
+        $errors = [];
+        $data = ['en' => ['title' => 'Hi'], 'es' => ['title' => 'hola']];
+        $args = [
+            null,
+            $data,
+            &$errors
```

Comment:
I needed before to attach translated entity errors to "parent entity". Now mergeTranslations return pair [entities, errors].